compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/FirHelpers.kt

```diff
@@ -321,3 +323,183 @@ val FirValueParameter.hasValOrVar: Boolean
         val source = this.source ?: return false
         return source.getChild(VAL_VAR_TOKEN_SET) != null
     }
+
+fun ConeKotlinType.canBeUsedForConstVal(): Boolean {
```

Comment:
Please create a separate file with functions related to constant expressions (this one, `checkConstantArguments`, `ConstantArgumentKind`, etc.). You can call this file `FirConstChecks.kt` or something like this.